tests/worker/test_reworker.py

```diff
@@ -346,7 +346,9 @@ def test_worker_and_data_events_produce_in_order(worker: Worker) -> None:
 def assert_running_count_plan_produces_ordered_worker_and_data_events(
     expected_events: list[WorkerEvent | DataEvent],
     worker: Worker,
-    task: Task = Task(name="count", params={"detectors": ["image_det"], "num": 1}),  # noqa: B008
+    task: Task = Task(
+        name="count", params={"detectors": ["image_det"], "num": 1}
+    ),  # noqa: B008
```

Comment:
Remove the noqa, make task: Task | None = None

Then in the body of the assert, task = task or Task(name="count", params={"detectors": ["image_det"], "num": 1})

Reply:
As is, task is instantiated when the method is first read, then every time it's called without an argument, the same task is being passed around, modified, run.